app/models/ability.rb

```diff
@@ -0,0 +1,19 @@
+class Ability
+  include CanCan::Ability
+
+  def initialize(user)
+    # Define abilities for the user here. For example:
+    user ||= User.new # Guest user (not logged in)
+
+    # Define abilities for different roles
+    if user.admin?
+      can :manage, :all # Admins can manage all resources
+    else
+      can :read, :all
+      can :create, Post
+      can :destroy, Post, author_id: user.id
+      can :destroy, Comment, author_id: user.id
```

Comment:
- [ ] good job in assigning roles to users, but from this your role, a user can not create a comment, to make it cleaner you can write like this 
```
can :manage, Post, author: user 
         can :manage, Comment, author: user
         can :read, :all
```
here a user can manage comments and posts if he is the owner and also read all

Reply:
- Well resolved 👍🏽